include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -0,0 +1,324 @@
+            (!stream.eof()))
+        {
+            stream.get(); // triggers error in stream and sets eof
+        }
+    }
+
+    //!\copydoc sequence_file_output_format_concept::write
+    template <typename stream_type,     // constraints checked by file
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void write(stream_type                          & stream,
+               sequence_file_output_options const   & options,
+               seq_type                             && sequence,
+               id_type                              && id,
+               qual_type                            && SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+
+        ranges::ostreambuf_iterator stream_it{stream};
+        size_t sequence_size = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size = ranges::size(sequence);
+
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing embl files."};
+        }
+        else
+        {
+            if (ranges::empty(id)) //[[unlikely]]
+                throw std::runtime_error{"The ID field may not be empty when writing embl files."};
+
+            if (options.complete_header)
+            {
+                ranges::copy(id, stream_it);
+            }
+            else
+            {
+                stream_it = 'I';
+                stream_it = 'D';
+                stream_it = ' ';
+                ranges::copy(id, stream_it);
+                stream_it = ';';
+                stream_it = ' ';
+                ranges::copy(std::to_string(sequence_size), stream_it);
+                ranges::copy(std::string_view{" BP.\n"}, stream_it);
+            }
+
+        }
+
+        // Sequence
+        if constexpr (detail::decays_to_ignore_v<seq_type>) // sequence
+        {
+            throw std::logic_error{"The SEQ and SEQ_QUAL fields may not both be set to ignore when writing "
+            "embl files."};
+        }
+        else
+        {
+            if (ranges::empty(sequence)) //[[unlikely]]
+                throw std::runtime_error{"The SEQ field may not be empty when writing embl files."};
+
+            ranges::copy(std::string_view{"SQ Sequence "}, stream_it);
+            ranges::copy(std::to_string(sequence_size), stream_it);
+            ranges::copy(std::string_view{" BP;"}, stream_it);
+            stream_it = '\n';
+            auto seq = sequence | ranges::view::chunk(60);
+            unsigned int i = 0;
+            size_t bp = 0;
+            while (bp < sequence_size)
+            {
+                ranges::copy(seq[i] | view::to_char
+                                    | ranges::view::chunk(10)
+                                    | ranges::view::join(' '), stream_it);
+                bp = std::min(sequence_size, bp + 60);
```

Comment:
instead of checking this while condition and accessing seq[i] I would recommend just iterating over the chunks